projects/eslint-plugin/rules/strict-tui-doc-example.js

```diff
@@ -0,0 +1,104 @@
+const getTypeName = require('../utils/get-type-name');
+
+const MESSAGE_ID = `strict-doc-example-extensions`;
+const ERROR_MESSAGE = `The import path extension should match the object key`;
+
+const DOC_EXAMPLE_INTERFACE_NAME = 'TuiDocExample';
+
+/**
+ * @type {TuiDocExample}
+ */
+const fileNameToExtension = {
+    TypeScript: '.ts',
+    HTML: '.html',
+    LESS: '.less',
+};
+
+/**
+ * Parses path into RegExp groups:
+ * 1. Everything before the last extension (greedy)
+ * 2. The last extension
+ * 3. Everything after the last extension (can be empty)
+ * @example
+ * // returns ['./examples/2/index.html/index', '.ts', '?raw']
+ * getPathGroups('./examples/2/index.html/index.ts?raw')
+ * @param {string} path
+ * @returns {RegExpMatchArray | null}
+ */
+const getPathGroups = path => path.match(/(.+)(\.(?:ts|less|scss|js|html))(.*)/);
+
+/**
+ * @type {import('eslint').Rule.RuleModule}
+ */
+module.exports = {
+    meta: {
+        type: 'problem',
+        docs: {description: ERROR_MESSAGE},
+        messages: {
+            [MESSAGE_ID]: ERROR_MESSAGE,
+        },
+        fixable: 'code',
+    },
+    create(context) {
+        return {
+            /**
+             * It would be better to support any object expression, not only within a class,
+             * but it's pretty complicated to extract the object type since all parents have different interfaces.
+             */
+            'PropertyDefinition[value.type="ObjectExpression"]'(node) {
+                if (getTypeName(node) !== DOC_EXAMPLE_INTERFACE_NAME) {
+                    return;
+                }
+
+                node.value.properties.forEach(prop => {
+                    // it can be either Identifier or Literal.
+                    const objKey = prop.key.name || prop.key.value;
+                    const objValue = prop.value;
+
+                    if (objValue.type !== 'ImportExpression') {
+                        return;
+                    }
+
+                    const {source} = objValue;
+
+                    /**
+                     * Extract an extension from the object key.
+                     * The object key can be a file name or module path.
+                     * At the same time the key can be invalid, that's why this value can be null.
+                     * @see TuiDocExample
+                     * @type {string|null}
+                     */
+                    const expectedExtension =
```

Comment:
@ddubrava What do you think about that case?

Can we show error for key where we try load resource by raw loader?

```ts
'./filesio.service': import('./examples/1/filesio.service.ts?raw'),
```

Reply:
ye, I think we can

Reply:
<img width="530" alt="Screenshot 2023-01-30 at 15 45 04" src="https://user-images.githubusercontent.com/22116465/215480639-100650fc-a6ee-45ae-84e2-b787a713b9fe.png">